src/tools/generators.ts

```diff
@@ -245,3 +259,279 @@ export function installGenerators(generators: string[]): string[] {
+    const messages = [
+      r.isMissing && "   • missing",
+      r.isInvalidSize && "   • wrong size (expected 1024x1024px)",
+    ].filter(Boolean)
+
+    if (messages.length) {
+      acc.push(`⚠️  ${r.fileName}:`, ...messages)
+    }
+
+    return acc
+  }, [])
+
+  return {
+    isValid: !validationMessages.length,
+    messages: validationMessages,
+  }
+}
+
+/**
+ * Generates app-icons for specified option.
+ */
+export async function generateAppIcons(option: `${Platforms}` | "all") {
+  const { path, exists, find, copy, write } = filesystem
+  const cwd = process.cwd()
+
+  const options = option === "all" ? Object.values(Platforms) : ([option] as `${Platforms}`[])
+
+  const optionGenerationSuccesses = []
+
+  // start the generation process for each platform
+  // looping instead of mapping allows us to await for each platform sequentially
+  for (const o of options) {
+    const optionProjectName = { ios: "iOS", android: "Android", expo: "Expo" }[o]
+
+    // find the output path for platform and check if it exists
+    // iOS is a bit weird since it's named differently for each project
+    const relativeOutputDirPath = {
+      expo: "assets/images",
+      android: "android/app/src/main/res",
+      ios:
+        (exists("ios") &&
+          find("ios", {
+            directories: true,
+            files: false,
+            matching: "AppIcon.appiconset",
+          })?.[0]) ||
+        "ios/**/Images.xcassets/AppIcon.appiconset",
+    }[o]
+    const outputDirPath = path(cwd, relativeOutputDirPath)
+
+    // if not, skip...
+    if (exists(outputDirPath) !== "dir") {
+      warning(
+        `⚠️  No output directory found for "${optionProjectName}" at "${outputDirPath}". Skipping...`,
+      )
+      continue
+    }
+
+    heading(`Generating ${optionProjectName} app icons...`)
+
+    const icons = APP_ICON_RULES.icons.filter((i) => i.platform === o)
+
+    // prepare each icon for generation sequentially
+    for (const i of icons) {
+      const inputFilePath = path(templatesDir(), "app-icon", i.inputFile)
+
+      // get the input file for sharp and do some initial transforms when necessary (e.g. radius and padding)
+      const inputFile = await (async function () {
+        if (!i.transform) return inputFilePath
+
+        try {
+          const { size, radius, padding } = i.transform
+          const cutoutMask = Buffer.from(
+            `<svg><rect x="0" y="0" width="${size}" height="${size}" rx="${radius}" ry="${radius}"/></svg>`,
+          )
+          return await sharp(inputFilePath)
+            .resize(size, size, { fit: "fill" })
+            .composite([{ input: cutoutMask, blend: "dest-in" }])
+            .extend({
+              top: padding,
+              bottom: padding,
+              left: padding,
+              right: padding,
+              background: { r: 0, g: 0, b: 0, alpha: 0 },
+            })
+            .toBuffer()
+        } catch (error) {}
+      })()
```

Comment:
then, for each icon for the selected option, we either return its path or if transforms are necessary, transform it into a buffer. In the case of Android legacy, we need to add padding and radius. Don't want to have developers do that manually in photoshop :)